x-pack/legacy/plugins/canvas/canvas_plugin_src/renderers/time_filter/components/datetime_input/index.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Moment } from 'moment';
+import { compose, withState, lifecycle } from 'recompose';
+import { DatetimeInput as Component, Props as ComponentProps } from './datetime_input';
+
+export interface Props {
+  /** Input value (Moment date object) */
+  moment?: Moment;
+  /** Function to invoke when the input changes */
+  setMoment: (m: Moment) => void;
+}
+
+export const DatetimeInput = compose<ComponentProps, Props>(
+  withState('valid', 'setValid', () => true),
+  withState('strValue', 'setStrValue', ({ moment }) =>
+    moment ? moment.format('YYYY-MM-DD HH:mm:ss') : ''
+  ),
+  lifecycle<Props & ComponentProps, {}>({
+    componentWillReceiveProps({ moment, setStrValue, setValid }) {
+      if (!moment) return;
+
+      if (this.props.moment && this.props.moment.isSame(moment)) {
+        return;
+      }
+      setStrValue(moment.format('YYYY-MM-DD HH:mm:ss'));
+      setValid(true);
+    },
+  })
```

Comment:
I kinda wanted to minimize this PRs footprint by only updating what needed to be updated. I removed recompose from `time_picker/index.js` since the typing in there with recompose was getting very hard to read. Def down for removing recompose in a future PR though